### PR TITLE
add current file name to error message

### PR DESCRIPTION
**Pull request by mpalpha**
Prepend the current file to the transformer error message.

### CODE REVIEW

**Comment by mpalpha** [comment]
should fix #163

File: lib/index.js [hunk 1]
```diff
@@ -57,7 +57,12 @@ function render({ filename, files, metadata, engineOptions }) {
 
       // Transform the contents
       debug(`rendering ${ext} extension for ${filename}`);
-      file.contents = transform.render(file.contents, engineOptions, locals).body;
+      try {
+        file.contents = transform.render(file.contents, engineOptions, locals).body;
+      } catch (err) {
+        err.message = `${filename}: ${err.message}`;
+        throw err;
```
Comment (ismay) [inline lib/index.js]: I'd prefer using the reject callback here instead of throwing (see the Promise constructor)

**Comment by ismay** [comment]
Hi @mpalpha, thanks for the PR! That's a good suggestion. Besides using the reject callback (see comment), would you mind opening a PR for the same change in `metalsmith-layouts`, and adding a test once we've resolved the outstanding issues? Thanks!

**Comment by mpalpha** [comment]
Error: The partial menu could not be found

to

Error: index.hbs: The partial menu could not be found

**Comment by ismay** [comment]
@mpalpha I had some time to look at this, and it looks good. However, I'd like to add tests to make sure it works as expected. However `getTransformer` would need to be mocked for that. Not that much trouble, but that will take a bit more time than I have at the moment. I'll revisit this asap and add some tests, and then we should be able to merge this and the related layouts PR.

**Pull request merged**